pkg/interceptors/results/results.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package results
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+	"google.golang.org/genproto/googleapis/api/httpbody"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+	"google.golang.org/protobuf/types/known/anypb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/transport"
+
+	v1alpha2pb "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"github.com/tektoncd/triggers/pkg/interceptors"
+)
+
+const (
+	// Service Account token path. See https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod
+	podTokenPath    = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	localAPIAddr    = "tekton-results-api-service.tekton-pipelines.svc.cluster.local:50051"
+	tlsSecretName   = "tekton-results-tls"
+	tektonNamespace = "tekton-pipelines"
+)
+
+var _ triggersv1.InterceptorInterface = (*Interceptor)(nil)
+
+// ErrInvalidContentType is returned when the content-type is not a JSON body.
+var ErrInvalidContentType = errors.New("form parameter encoding not supported, please change the hook to send JSON payloads")
+
+type Interceptor struct {
+	KubeClientSet kubernetes.Interface
+	Logger        *zap.SugaredLogger
+	ResultClient  v1alpha2pb.ResultsClient
+}
+
+func NewInterceptor(k kubernetes.Interface, l *zap.SugaredLogger) *Interceptor {
+	return &Interceptor{
+		Logger:        l,
+		KubeClientSet: k,
+	}
+}
+
+func (w *Interceptor) Process(ctx context.Context, r *triggersv1.InterceptorRequest) *triggersv1.InterceptorResponse {
+	headers := interceptors.Canonical(r.Header)
+	if v := headers.Get("Content-Type"); v == "application/x-www-form-urlencoded" {
+		return interceptors.Fail(codes.InvalidArgument, ErrInvalidContentType.Error())
+	}
+
+	p := triggersv1.ResultsInterceptor{}
+	if err := interceptors.UnmarshalParams(r.InterceptorParams, &p); err != nil {
+		return interceptors.Failf(codes.InvalidArgument, "failed to parse interceptor params: %v", err)
+	}
+	if p.APIAddr == "" {
+		p.APIAddr = localAPIAddr
+	}
+	if w.ResultClient == nil {
```

Comment:
the way that this is implemented, once the ResultClient is set on the interceptor service, it won't ever reconfigure the API server address or token. Given you have these as a parameter for triggers to set, i don't think the resultclient should be a property of the interceptor itself.